### PR TITLE
If locality-based exclusions are enabled, only make use of the locality

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Otherwise we might issue an exclusions with the IP address.

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)

## Discussion

-

## Testing

I will add some additional unit tests for this behaviour.

## Documentation

-

## Follow-up

-

### CODE REVIEW

**Review by johscheuer — COMMENTED** [review]
This change is blocked on a new release containing: https://github.com/apple/foundationdb/pull/10976. We have to add additional version checks to make sure locality based exclusions are only used for this version that has the fix.

File: internal/removals/remove.go [hunk 1]
```diff
@@ -126,15 +126,18 @@ func GetRemainingMap(logger logr.Logger, adminClient fdbadminclient.AdminClient,
 			continue
 		}
 
-		if len(processGroup.Addresses) == 0 {
-			logger.Info("Getting remaining removals to check for exclusion", "processGroupID", processGroup.ProcessGroupID, "reason", "missing address")
+		// If we use localities for exclusions we don't have to care about the addresses.
+		if cluster.UseLocalitiesForExclusion() {
```
Comment (johscheuer) [inline internal/removals/remove.go:130]: The use localities for exclusion must have an additional check for the released version.

Comment (johscheuer) [reply internal/removals/remove.go:130]: We have to add the IP addresses for log processes, in case they might be missing in the status.

**Comment by johscheuer** [comment]
This PR: https://github.com/FoundationDB/fdb-kubernetes-operator/pull/1868 add the version check.

**Review by AnnigeriShambu — APPROVED** [review]

**Pull request merged**